keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ArtifactInterfaces.kt

```diff
@@ -0,0 +1,82 @@
+package com.netflix.spinnaker.keel.api
+
+import com.netflix.spinnaker.keel.api.artifacts.ArtifactType
+
+/**
+ * Implemented by [ResourceSpec] or concrete resource types that (may) contain artifact information, typically compute
+ * resources.
+ *
+ * The fields on this interface are nullable because typically the spec will _not_ have that information available
+ * before the corresponding [ResourceHandler] has resolved the resource.
+ */
+interface ArtifactProvider {
+  val artifactName: String?
+  val artifactType: ArtifactType?
```

Comment:
Should this have some json annotations on it to hide it, or no? right now when you query your delivery config this field comes back on the cluster resource

Reply:
The corresponding fields are annotated in the implementing classes. Can't annotate here because we don't want to depend on jackson in `keel-api` due to the plugin stuff.

Reply:
You can use a mixin like in `MonikeredMixin`

Reply:
I tried that after reading your comment but thought that the end result was much more confusing, as the resource spec types already have jackson annotations to indicate which fields to omit, and handling just a sub-set of those via mixins makes it entirely unclear that certain fields will be omitted by just looking at those classes.

Reply:
@emjburns Sorry, I may have misinterpreted your original question. The existing `imageProvider` property in `ClusterSpec` is not annotated, and the newly added `containerProvider` property in `TitusClusterSpec` is not annotated either, so you still get that info in your delivery config:

Example:
```yaml
name: "fnord-manifest"
application: "fnord"
serviceAccount: "delivery-engineering@netflix.com"
artifacts:
- name: "fnord"
  reference: "fnord"
  statuses: []
  type: "deb"
environments:
- name: "test"
  resources:
  - kind: "ec2/cluster@v1"
    metadata:
      id: "ec2:cluster:testaccount:fnord-test"
      uid: "01E3WMPAPYQN6Z5QH0A6MMF81B"
      application: "fnord"
      serviceAccount: "delivery-engineering@netflix.com"
    spec:
      moniker:
        app: "fnord"
        stack: "test"
      imageProvider:
        reference: "fnord"
      # ...
```